BLE_GattServer_ExperimentalServices/source/main.cpp

```diff
@@ -0,0 +1,179 @@
+            _ble(ble),
+            _event_queue(event_queue),
+            _chainable_gap_event_handler(chainable_gap_event_handler),
+            _link_loss_service(_ble, _event_queue, _chainable_gap_event_handler),
+            _adv_data_builder(_adv_buffer)
+    {
+    }
+
+    void start()
+    {
+        _ble.init(this, &LinkLossDemo::on_init_complete);
+
+        _event_queue.dispatch_forever();
+    }
+
+private:
+    void on_init_complete(BLE::InitializationCompleteCallbackContext *params)
+    {
+        if (params->error != BLE_ERROR_NONE) {
+            printf("Ble initialization failed.");
+            return;
+        }
+
+        print_mac_address();
+
+        _chainable_gap_event_handler.addEventHandler(this);
+        _ble.gap().setEventHandler(&_chainable_gap_event_handler);
+
+        _link_loss_service.set_event_handler(this);
+        _link_loss_service.set_timeout(60000ms);
+
+        ble_error_t error = _link_loss_service.init();
+
+        if (error) {
+            printf("Link Loss Service could not be initialized");
+            return;
+        }
+
+        start_advertising();
+    }
+
+    void start_advertising()
+    {
+        UUID link_loss_service_uuid = GattService::UUID_LINK_LOSS_SERVICE;
+
+        ble::AdvertisingParameters adv_parameters(
+                ble::advertising_type_t::CONNECTABLE_UNDIRECTED,
+                ble::adv_interval_t(ble::millisecond_t(100))
+        );
+
+        _adv_data_builder.setFlags();
+        _adv_data_builder.setAppearance(ble::adv_data_appearance_t::UNKNOWN);
+        _adv_data_builder.setLocalServiceList({&link_loss_service_uuid, 1});
+        _adv_data_builder.setName(DEVICE_NAME);
+
+        ble_error_t error = _ble.gap().setAdvertisingParameters(
+                ble::LEGACY_ADVERTISING_HANDLE,
+                adv_parameters
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingParameters() failed\r\n");
+            return;
+        }
+
+        error = _ble.gap().setAdvertisingPayload(
+                ble::LEGACY_ADVERTISING_HANDLE,
+                _adv_data_builder.getAdvertisingData()
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingPayload() failed\r\n");
+            return;
+        }
+
+        error = _ble.gap().startAdvertising(ble::LEGACY_ADVERTISING_HANDLE);
+
+        if (error) {
+            printf("_ble.gap().startAdvertising() failed\r\n");
+            return;
+        }
+
+        printf("Device advertising, please connect\r\n");
+    }
+
+    void on_alert_requested(LinkLossService::AlertLevel level) final
+    {
+        if (level == LinkLossService::AlertLevel::MILD_ALERT) {
+            printf("%s\r\n", MILD_ALERT);
+        } else {
+            printf("%s\r\n", HIGH_ALERT);
+        }
+    }
+
+    void on_alert_end() final
+    {
+        printf("Alert timed out\r\n");
```

Comment:
```suggestion
        printf("Alert ended\r\n");
```
alert may end for other reason, like the peer re-connected